draft-wang-ppm-dap-taskprov.md

```diff
@@ -268,22 +304,73 @@ unimplemented DP mechanism).
+describes a mechanism whose  key feature is that task configuration is
+performed completely in-band, via HTTP request headers.
 
-~~~
-task_id = SHA-256(task_config)
-~~~
+This method presumes the existence of a logical "task author" (written as
+"Author" hereafter) who is capable of pushing configurations to Clients. All
+parameters required by downstream entities (the Aggregators and Collector) are
+carried by HTTP headers piggy-backed on the protocol flow.
 
-where `task_config` is the `TaskConfig` structure disseminated by the Author.
-Function SHA-256() is as defined in {{SHS}}.
+This mechanism is designed with the same security and privacy considerations of
+the core DAP protocol. The Author is not regarded as a trusted third party: it
+is incumbent on all protocol participants to verify the task configuration
+disseminated by the Author and opt-out if the parameters are deemed insufficient
+for privacy. In particular, adopters of this mechanism should presume the
+Author is under the adversary's control. In fact, we expect in a real-world
+deployment that the Author may be co-located with the Collector.
+
+The DAP protocol also requires configuring the entities with a variety of assets
+that are not task-specific, but are important for establishing
+Client-Aggregator, Collector-Aggregator, and Aggregator-Aggregator
+relationships. These include:
+
+* The Collector's HPKE {{!RFC9180}} configuration used by the Aggregators to
+  encrypt aggregate shares.
+
+* Any assets required for authenticating HTTP requests.
+
+This section does not specify a mechanism for provisioning these assets; as in
+the core DAP protocol; these are presumed to be configured out-of-band.
+
+Note that we consider the VDAF verification key {{!VDAF}}, used by the
+Aggregators to aggregate reports, to be a task-specific asset. This document
+specifies how to derive this key for a given task from a pre-shared secret,
+which in turn is presumed to be configured out-of-band.
+
+## Overview
+
+The process of provisioning a task begins when the Author disseminates the task
+configuration to the Collector and each of the Clients. When a Client issues an
+upload request to the Leader (as described in {{Section 4.3 of !DAP}}), it
+includes in an HTTP header the task configuration it used to generate the
+report. We refer to this process as "task advertisement". Before consuming the
+report, the Leader parses the configuration and decides whether to opt-in; if
+not, the task's execution halts.
+
+Otherwise, if the Leader does opt-in, it advertises the task to the Helper
+during the aggregation protocol ({{Section 4.4 of !DAP}}). In particular, it
+includes the task configuration in an HTTP header of each aggregation job
+request for that task. Before proceeding, the Helper must first parse the
+configuration and decide whether to opt-in; if not, the task's execution halts.
+
+## Task Advertisement
+
+To advertise a task to its peer, a Taskbind participant includes a header
```

Comment:
This should be Taskprov participant?

Reply:
Good catch. This appears to be leftovers from `%s/Taskprov/Taskbind/`. I meant "protocol" here.